app/services/foreman_ansible/inventory_creator.rb

```diff
@@ -97,7 +97,7 @@ def ansible_extra_options(host)
 
     def remote_execution_options(host)
       params = {
-        'ansible_user' => host_setting(host, 'remote_execution_ssh_user'),
+        'ansible_user' => @template_invocation.job_invocation&.ssh_user || host_setting(host, 'remote_execution_ssh_user'),
```

Comment:
If I'm reading this right, then `@template_invocation` can be `nil` which could then raise. I know it is sort of guarded on L55, but it would be nice if we could stay on the safe side of things.

Reply:
Should be better now!